web3/_utils/rpc_abi.py

```diff
@@ -51,6 +51,7 @@
     'eth_sendRawTransaction': ['bytes'],
     'eth_sendTransaction': TRANSACTION_PARAMS_ABIS,
     'eth_sign': ['address', 'bytes'],
+    'eth_submitHashrate': ['uint', None],
```

Comment:
I think this is right (not sure if it's always `bytes32`)
```suggestion
    'eth_submitHashrate': ['uint', 'bytes'],
```
Also, maybe `['bytes8', 'bytes32', 'bytes32']` for `eth_submitWork`?

Adding this extra fields should automatically handle arguments like submitting hex strings, hex literals, or python `bytes`, as well as length checks.

Reply:
[geth](https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_submithashrate) and [parity](https://wiki.parity.io/JSONRPC-eth-module#eth_submithashrate) both specify 32 bytes, but I don't think the EIP does  ¯\_(ツ)_/¯

Reply:
Eh, the EIP is still a draft. Since geth & parity implement the same way, that wins over the doc. :)